docs/providers/aws/guide/serverless.yml.md

```diff
@@ -16,117 +16,259 @@ layout: Doc
+  stackName: custom-stack-name
+  # Use a custom name for the API Gateway API
+  apiName: custom-api-name
+  # Use a custom name for the websockets API
+  websocketsApiName: custom-websockets-api-name
+  # custom route selection expression
+  websocketsApiRouteSelectionExpression: $request.body.route
+  # optional, if set to 'true', disables default behavior of generating export names for CloudFormation outputs
+  disableDefaultOutputExportNames: false
+  # Optional CloudFormation tags to apply to all services and functions
+  tags:
+    foo: bar
+    baz: qux
+  # Optional CloudFormation tags to apply to the stack
+  stackTags:
+    key: value
+  # List of existing Amazon SNS topics in the same region where notifications about stack events are sent.
+  notificationArns:
+    - 'arn:aws:sns:us-east-1:XXXXXX:mytopic'
+  stackParameters:
+    - ParameterKey: 'Keyname'
+      ParameterValue: 'Value'
+  # Disable automatic rollback by CloudFormation on failure. To be used for non-production environments.
+  disableRollback: true
+  rollbackConfiguration:
+    MonitoringTimeInMinutes: 20
+    RollbackTriggers:
+      - Arn: arn:aws:cloudwatch:us-east-1:000000000000:alarm:health
+        Type: AWS::CloudWatch::Alarm
+      - Arn: arn:aws:cloudwatch:us-east-1:000000000000:alarm:latency
+        Type: AWS::CloudWatch::Alarm
+  tracing:
+    # Can only be true if API Gateway is inside a stack.
+    apiGateway: true
+    # Optional, can be true (true equals 'Active'), 'Active' or 'PassThrough'
+    lambda: true
+```
+
+### General function settings
+
+Some function settings can be defined for all functions inside the `provider` key:
+
+```yml
+# serverless.yml
+
+provider:
   runtime: nodejs14.x
-  stage: dev # Default stage to be used. Default is "dev"
-  region: us-east-1 # Default region to be used. Default is "us-east-1"
-  stackName: custom-stack-name # Use a custom name for the CloudFormation stack
-  apiName: custom-api-name # Use a custom name for the API Gateway API
-  websocketsApiName: custom-websockets-api-name # Use a custom name for the websockets API
-  websocketsApiRouteSelectionExpression: $request.body.route # custom route selection expression
-  profile: production # The default profile to use with this service
-  memorySize: 512 # Overwrite the default memory size. Default is 1024
-  timeout: 10 # The default is 6 seconds. Note: API Gateway current maximum is 30 seconds
-  logRetentionInDays: 14 # Set the default RetentionInDays for a CloudWatch LogGroup
-  kmsKeyArn: arn:aws:kms:us-east-1:XXXXXX:key/some-hash # KMS key arn which will be used for encryption for all functions
+  # Default memory size for functions (default: 1024MB)
+  memorySize: 512
+  # Default timeout for functions (default: 6 seconds)
+  # Note: API Gateway has a maximum timeout of 30 seconds
+  timeout: 10
+  # Function environment variables
+  environment:
+    APP_ENV_VARIABLE: FOOBAR
+  # Duration for CloudWatch log retention (default: forever)
```

Comment:
Can I get a confirmation on the "default: forever"?

Reply:
the default is the CloudFormation/AWS default - we don't set any specific value if it's not provided, so based on CW docs it's "forever" https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html